components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/handler/impl/OAuth2AccessTokenHandler.java

```diff
@@ -283,4 +290,24 @@ private boolean isSSOSessionBasedTokenBinding(String tokenBindingType) {
 
         return SSO_SESSION_BASED_TOKEN_BINDER.equals(tokenBindingType);
     }
+
+    /**
+     * Set the service provider details to a thread local variable to be consumed by the provisioning framework.
+     *
+     * @param oauthAppConsumerKey           Client ID of the OAuth client application.
+     * @param serviceProviderTenantDomain   Tenant Domain of the OAuth application.
+     */
+    private void setProvisioningServiceProviderThreadLocal(String oauthAppConsumerKey,
+                                                           String serviceProviderTenantDomain) {
+
+        if (serviceProviderTenantDomain != null) {
+            ThreadLocalProvisioningServiceProvider provisioningServiceProvider =
+                    new ThreadLocalProvisioningServiceProvider();
+            provisioningServiceProvider.setServiceProviderName(oauthAppConsumerKey);
+            provisioningServiceProvider.setServiceProviderType(ProvisioningServiceProviderType.OAUTH);
+            provisioningServiceProvider.setClaimDialect(DEFAULT_SCIM2_DIALECT);
```

Comment:
Do we need to set the dialect?

Reply:
Removed in https://github.com/wso2-extensions/identity-carbon-auth-rest/pull/132/commits/7d66bbadf2e1d85c33ca72ea6fec2a0b5ea38e9f